src/dstack/_internal/core/backends/gcp/compute.py

```diff
@@ -87,9 +92,24 @@ def terminate_instance(
             backend_data_dict = json.loads(backend_data)
             zone = backend_data_dict["zone"]
         try:
-            self.instances_client.delete(
-                project=self.config.project_id, zone=zone, instance=instance_id
-            )
+            if "tpu" in instance_id:
+                name = f"projects/{self.project_id}/locations/{zone}/nodes/{instance_id}"
+                delete_request = tpu_v2.DeleteNodeRequest(
+                    name=name,
+                )
+                operation = self.tpu_client.delete_node(request=delete_request)
+                while not operation.done():
```

Comment:
Why not gcp_resources.wait_for_operation()?

Reply:
I will use gcp_resources.wait_for_operation(). This will make it consistent.